compiler/src/iree/compiler/Codegen/Dialect/GPU/TargetUtils/ConfigUtils.cpp

```diff
@@ -238,9 +240,23 @@ getMatmulLoweringConfigAndWorkgroupSize(SmallVector<int64_t> bounds,
       nDims.back() !=
       llvm::cast<AffineDimExpr>(maps[1].getResults().back()).getPosition();
 
+  bool mustBeAligned = true;
   std::optional<GPUMMASchedule> schedule = getMmaScheduleFromProblemAndTarget(
       target, problem, transposedLhs, transposedRhs);
 
+  // TODO (nirvedhmeshram, jerryyin): Support all GEMM types.
+  // TODO (nirvedhmeshram): Support fused leading op.
```

Comment:
Just haven't tested it. It might not. I can take a quick look at a IGEMM case with unaligned shape and see how it goes as IIUC that has a fused producer.

Reply:
Another reason was that without this check it will run by default as IGEMM is on by default and I am not sure we are ready on this path to support all shapes.

Reply:
Ya that didnt work here is the [dump](https://gist.github.com/nirvedhmeshram/e56181182ffdc5a6ed8d674ce838a56d).
 From what I noticed I think the issue is that the producer nis not padded so it leads to some dynamic allocations

Reply:
ok actually the above dump was just me not padding the reduction dim enough, after fixing that there is one dynamic alloca thats blocking us from running this now
here is the [dump](https://gist.github.com/nirvedhmeshram/52ec869213db171414abf0d346f159c3)

Reply:
I see, one step at a time sounds good then. We can figure out what's going wrong later.